vyper/parser/pre_parser.py

```diff
@@ -49,6 +52,9 @@ def pre_parse(code):
             # Prevent semi-colon line statements.
             elif (token.type, token.string) == (OP, ";"):
                 raise StructureException("Semi-colon statements not allowed.", token.start)
+            # Prevent use of None literal
```

Comment:
This blocks `None` at the tokenization stage, but a user might want to make a contract or function named `None` (weird, I know). I'm thinking this check should be moved to a later stage - https://github.com/ethereum/vyper/blob/921bb80cf59c9be0067ed20e1a7b6ad0b59598b3/vyper/parser/expr.py#L147.

Reply:
@charles-cooper Yeah so that's actually intentional haha. 😅I previously had this set up to block at a later stage, when checking assignments or comparisons. Refactored though because it just generally seems like something to prevent a user from doing even if they want to, because it is confusing given that Vyper is based mostly on Python's syntax and would allow for potentially very misleading code (where one thinks a custom type or custom struct named `None` is the Python `None` but its actually some custom thing called `None`.

Reply:
I think the best thing in that case would be if we just add `None` as 'n reserved keyword?

Reply:
@jacqueswww That makes sense, just pushed a commit that implements that approach instead.